codegen/src/wasm/host.rs

```diff
@@ -88,3 +90,77 @@ pub enum CompilerLabel {
     ReserveMemory32,
     ReserveMemory64,
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_addmod_mulmod_host_functions() {
+        // Test ADDMOD host function conversion
+        let addmod_func = HostFunc::try_from(("zinkc", "u256_addmod"));
+        assert!(addmod_func.is_ok());
+        let addmod_host_func = addmod_func.unwrap();
+
+        // Verify it maps to the correct EVM opcode
+        match addmod_host_func {
+            HostFunc::Evm(OpCode::ADDMOD) => {}
+            _ => panic!("Expected ADDMOD opcode"),
+        }
+
+        // Verify stack input/output for ADDMOD
+        assert_eq!(addmod_host_func.stack_in(), 3);
+        assert_eq!(addmod_host_func.stack_out(), 1);
+
+        // Test MULMOD host function conversion
+        let mulmod_func = HostFunc::try_from(("zinkc", "u256_mulmod"));
+        assert!(mulmod_func.is_ok());
+        let mulmod_host_func = mulmod_func.unwrap();
+
+        // Verify it maps to the correct EVM opcode
+        match mulmod_host_func {
+            HostFunc::Evm(OpCode::MULMOD) => {}
+            _ => panic!("Expected MULMOD opcode"),
+        }
+
+        // Verify stack input/output for MULMOD
+        assert_eq!(mulmod_host_func.stack_in(), 3);
+        assert_eq!(mulmod_host_func.stack_out(), 1);
+    }
+
+    #[test]
+    fn test_addmod_mulmod_example_scenarios() {
+        fn modular_add(a: u64, b: u64, n: u64) -> u64 {
+            ((a % n) + (b % n)) % n
+        }
+
+        let test_cases = [
+            (10u64, 20, 7, 2, 4),
+            (100, 200, 3, 0, 2),
+            (5, 6, 7, 4, 2),
+            (10, 11, 12, 9, 2),
+        ];
+
+        for (a, b, n, expected_add, expected_mul) in test_cases.iter() {
+            let actual_add = modular_add(*a, *b, *n);
+            let actual_mul = (*a * *b) % *n;
+
+            println!(
+                "Debug: a={}, b={}, N={}, actual_add={}, expected_add={}",
+                a, b, n, actual_add, expected_add
+            );
+
+            assert_eq!(
+                actual_add, *expected_add,
+                "ADDMOD failed for inputs {}, {}, {} (got {}, expected {})",
+                a, b, n, actual_add, expected_add
+            );
+
+            assert_eq!(
+                actual_mul, *expected_mul,
+                "MULMOD failed for inputs {}, {}, {} (got {}, expected {})",
+                a, b, n, actual_mul, expected_mul
+            );
+        }
+    }
```

Comment:
```suggestion
```

hmm this test does not make sense, you are actually testing your `modular_add`, instead, we need an implementation in real zink contracts, see [test_addmod](https://github.com/zink-lang/zink/commit/339daebeb1a117f3b0ee571521a7d4fb5e987926#diff-d6877096bf34e2026074b85f2fb8fa6ef1d706363c4bbd03cd6d4f8e28d01bdaR20) for example